src/withEffects.js

```diff
@@ -62,13 +62,20 @@ function runIfEffect(actions, currentEvent, maybeEffect) {
       case HTTP:
         props.options = props.options || {}
         props.options.response = props.options.response || "json"
+        var errorAction = props.options.error || props.action
+        if (props.options.error) {
+          delete props.options.error
+        }
         fetch(props.url, props.options)
           .then(function(response) {
             return response[props.options.response]()
           })
           .then(function(result) {
             getAction(actions, props.action)(result)
           })
+          .catch(function(err) {
+            getAction(actions, errorAction)(err)
```

Comment:
In the case where `errorAction` is `props.action`, shouldn't you call `getAction(actions, errorAction)({ error })` instead? 
According to the requirements in #15 at least.

Reply:
Would you say: `getAction(actions, errorAction)({ error: err })` ?
I don't think so. I don't think it's a good idea to "transform" the result returned by `fetch` before sending it to the user function (handler).
I would not "the catch" to introduce breaking changes or anything.
Before this merge request, when a ajax request failed,  `getAction(actions, props.action)(result)` was called with the error as result. The new version does exactly the same.

What requirements are you talking about ?
Two implementations are provided by @okwolf. It looks like we adopted the first one. We can so ignore the second option. 

Maybe I didn't understand your review :octocat: Thanks for reviewing :smiley:

Reply:
Ah yes, I miss-read the issue, sorry :/

Although I still have an issue with this, in the case where the user doesn't provide an `options.error`, the line `getAction(actions, errorAction)(err)` calls the user action with the error, instead of with the result of the fetch. How is the user supposed to differentiate between a successful fetch and a failed one? Like this?
```javascript
const actions = {
login: form => http(
    "/login",
    "loginComplete",
    {
      method: "POST",
      body: form
    }
  ),
  loginComplete: () => data => {
    if(data instanceof Error) {
      // error
    } else {
      // success
    }
  }
}
```

Reply:
Depending of what fetch returns, yes. And check the `instanceof` or if a key is in the object or whatever. The same behavior as before. 
If the user wants to check the error without the tool we offer, it's up to him to get over it! :)

(The same way fetch does: if you don't provide the `.catch`, the error comes in the `.then` and it's up to you to check if the data is error or not)

Reply:
Ok, sounds good to me.
Thanks for the explanations!